tiledb/api/src/query/write/input/arrow.rs

```diff
@@ -0,0 +1,753 @@
+        let validity = validity_buffer(self, is_nullable)?;
+
+        Ok(QueryBuffers {
+            data,
+            cell_structure,
+            validity,
+        })
+    }
+}
+
+impl DataProvider for LargeStringArray {
+    type Unit = u8;
+
+    fn query_buffers(
+        &self,
+        cell_val_num: CellValNum,
+        is_nullable: bool,
+    ) -> TileDBResult<QueryBuffers<Self::Unit>> {
+        let cell_structure = cell_structure_var(self.offsets(), cell_val_num)?;
+        let data = Buffer::Borrowed(self.value_data());
+        let validity = validity_buffer(self, is_nullable)?;
+
+        Ok(QueryBuffers {
+            data,
+            cell_structure,
+            validity,
+        })
+    }
+}
+
+impl TypedDataProvider for FixedSizeListArray {
+    fn typed_query_buffers(
+        &self,
+        cell_val_num: CellValNum,
+        is_nullable: bool,
+    ) -> TileDBResult<TypedQueryBuffers> {
+        let cell_structure = cell_structure_fixed(
+            self.value_length(),
+            self.len(),
+            cell_val_num,
+        )?;
+
+        let validity = validity_buffer(self, is_nullable)?;
+        apply_to_list_element(
+            self.value_type(),
+            self.values(),
+            cell_structure,
+            validity,
+        )
+    }
+}
+
+impl TypedDataProvider for GenericListArray<i64> {
+    fn typed_query_buffers(
+        &self,
+        cell_val_num: CellValNum,
+        is_nullable: bool,
+    ) -> TileDBResult<TypedQueryBuffers> {
+        let cell_structure = cell_structure_var(self.offsets(), cell_val_num)?;
+        let validity = validity_buffer(self, is_nullable)?;
+
+        apply_to_list_element(
+            self.value_type(),
+            self.values(),
+            cell_structure,
+            validity,
+        )
+    }
+}
+
+impl TypedDataProvider for dyn ArrowArray {
+    fn typed_query_buffers(
+        &self,
+        cell_val_num: CellValNum,
+        is_nullable: bool,
+    ) -> TileDBResult<TypedQueryBuffers> {
+        let c = cell_val_num;
+        let n = is_nullable;
+
+        use arrow::datatypes::{DataType as ADT, *};
+        match self.data_type() {
+            ADT::Null
+            | ADT::Boolean
+            | ADT::Float16
+            | ADT::Duration(_)
+            | ADT::Interval(_)
+            | ADT::Binary
+            | ADT::Utf8
+            | ADT::BinaryView
+            | ADT::Utf8View
+            | ADT::List(_)
+            | ADT::ListView(_)
+            | ADT::LargeListView(_)
+            | ADT::Struct(_)
+            | ADT::Union(_, _)
+            | ADT::Dictionary(_, _)
+            | ADT::Decimal128(_, _)
+            | ADT::Decimal256(_, _)
+            | ADT::Map(_, _)
+            | ADT::RunEndEncoded(_, _) => {
+                Err(Error::InvalidArgument(anyhow!(format!(
+                    "Unsupported Arrow datatype as query input: {}",
+                    self.data_type()
+                ))))
```

Comment:
Was there a particular reason for enumerating all of these instead of having a `_ =>` default arm? It doesn't bother me either direction, I just can't come up with a reason other than just trying to be specific in case this list changes in the future.

Reply:
I wrote it down this way before filling in the error.  I wanted to make sure I didn't miss anything.

Now that I haven't, we could change it, but my inclination is to leave it as is just to be extra sure we catch everything if arrow introduces a new type.